### PR TITLE
Delivery

### PR DESCRIPTION
**Pull request by Sobi1995**

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -3,7 +3,10 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
   </PropertyGroup>
   <ItemGroup>
+    <PackageVersion Include="Microsoft.Extensions.Hosting" Version="8.0.0" />
     <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+    <PackageVersion Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.19.6" />
+    <PackageVersion Include="RabbitMQ.Client" Version="6.6.0" />
```
Comment (behdadkd) [inline src/Directory.Packages.props:9]: use 6.8.1 veresion

File: src/Services/Core/Core.Api/Core.Api.csproj [hunk 2]
```diff
@@ -4,6 +4,10 @@
 		<TargetFramework>net8.0</TargetFramework>
 		<Nullable>enable</Nullable>
 		<ImplicitUsings>enable</ImplicitUsings>
+		<UserSecretsId>e4afb17d-f6a5-4142-a11c-96776823af2d</UserSecretsId>
```
Comment (behdadkd) [inline src/Services/Core/Core.Api/Core.Api.csproj:7]: remove user secret

File: src/Services/Core/Core.Api/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#See https://aka.ms/customizecontainer to learn how to customize your debug container and how Visual Studio uses this Dockerfile to build your images for faster debugging.
```
Comment (behdadkd) [inline src/Services/Core/Core.Api/Dockerfile:1]: merg main branch this is duplicate

File: src/Services/Delivery/Delivery.csproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk.Worker">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <UserSecretsId>dotnet-Delivery-da584e9e-1e55-4e03-8bf7-370f49f7f3f7</UserSecretsId>
```
Comment (behdadkd) [inline src/Services/Delivery/Delivery.csproj:7]: remove user secret

File: src/Services/Delivery/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#See https://aka.ms/customizecontainer to learn how to customize your debug container and how Visual Studio uses this Dockerfile to build your images for faster debugging.
+
+FROM mcr.microsoft.com/dotnet/runtime:8.0 AS base
```
Comment (behdadkd) [inline src/Services/Delivery/Dockerfile:3]: make it alpine

File: src/Services/Delivery/Program.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+using System.Text;
+
+var builder = Host.CreateApplicationBuilder(args);
+builder.Services.AddHostedService<Worker>();
+
+
+var factory = new ConnectionFactory { HostName = "rabbitmq", Port = 5672 };
+var connection = factory.CreateConnection();
+using var channel = connection.CreateModel();
+
```
Comment (behdadkd) [inline src/Services/Delivery/Program.cs:13]: declare queue 
you can see example in behdadkd/DeleveryEngine Engine2

File: src/Services/Delivery/Program.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+var consumer = new EventingBasicConsumer(channel);
+consumer.Received += (model, eventArgs) =>
+{
+    var body = eventArgs.Body.ToArray();
+    var message = Encoding.UTF8.GetString(body);
+    Console.WriteLine(message);
```
Comment (behdadkd) [inline src/Services/Delivery/Program.cs:19]: make a adaptor that make a 5 second task.delay and call it here insted of Console.WriteLine

File: src/Services/Delivery/Worker.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
+namespace Delivery;
```
Comment (behdadkd) [inline src/Services/Delivery/Worker.cs:1]: use a adapter and call it in +Recived on Cunsomer 
insted of this

File: src/Services/Delivery/Program.cs [hunk 9]
```diff
@@ -0,0 +1,27 @@
+using Delivery;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Text;
+
+var builder = Host.CreateApplicationBuilder(args);
+builder.Services.AddHostedService<Worker>();
+
+
+var factory = new ConnectionFactory { HostName = "rabbitmq", Port = 5672 };
```
Comment (behdadkd) [inline src/Services/Delivery/Program.cs:10]: use conection string from IConfiguration 
See "ProcessNotificationJob"

File: src/docker-compose.override.yml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+version: '3.4'
```
Comment (behdadkd) [inline src/docker-compose.override.yml:1]: remove this file

**Review by behdadkd — CHANGES_REQUESTED** [review]
❤️

File: .vscode/launch.json
Comment (MohammadMokhtarii) [inline .vscode/launch.json:1]: .vscode folder must be ignored via .gitignore

File: src/Services/Core/Core.Api/Core.Api.csproj [hunk 11]
```diff
@@ -1,9 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
 	<PropertyGroup>
-		<DockerfileContext>..\..\..</DockerfileContext>
```
Comment (MohammadMokhtarii) [inline src/Services/Core/Core.Api/Core.Api.csproj:4]:
```
<DockerfileContext>..\..\..</DockerfileContext>
<DockerComposeProjectPath>..\..\..\docker-compose.dcproj</DockerComposeProjectPath>
<InvariantGlobalization>false</InvariantGlobalization>
```

Shouldnt be Deleted For docker compose config in vs ide

File: src/Services/Delivery/Core/ProcessNotificationJob.cs [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Delivery.Core;
+
+public class ProcessNotificationJob : BackgroundService
+{
+    private readonly IConfiguration _configuration;
+    private readonly IPushNotificationAdapter _pushNotificationAdapter;
+
+    private IConnection _connection;
+    private IModel _channel;
```
Comment (MohammadMokhtarii) [inline src/Services/Delivery/Core/ProcessNotificationJob.cs:13]: Channel Must be Declare For Every  AsyncEventingBasicConsumer InOrder to work properly if you are going to bind multiple consumer to queue

File: src/Services/Delivery/Core/ProcessNotificationJob.cs [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    public ProcessNotificationJob(IConfiguration configuration, IPushNotificationAdapter pushNotificationAdapter)
+    {
+        _configuration = configuration;
+        _pushNotificationAdapter = pushNotificationAdapter;
+        CreateConnectionFactory();
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        foreach (var number in Enumerable.Range(1, 10))
```
Comment (MohammadMokhtarii) [inline src/Services/Delivery/Core/ProcessNotificationJob.cs:25]: use fore instead of foreach

File: src/Services/Delivery/Delivery.csproj [hunk 14]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Worker">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+ 
+  </PropertyGroup>
```
Comment (MohammadMokhtarii) [inline src/Services/Delivery/Delivery.csproj:8]:
```
<PropertyGroup>
<TargetFramework>net8.0</TargetFramework>
<Nullable>enable</Nullable>
<ImplicitUsings>enable</ImplicitUsings>

</PropertyGroup>
```


Should be removed becasue it has been configed globally in directory.build.props

File: src/Services/Delivery/appsettings.json [hunk 15]
```diff
@@ -0,0 +1,11 @@
+{
+  "ConnectionStrings": {
```
Comment (MohammadMokhtarii) [inline src/Services/Delivery/appsettings.json:2]: declare below config in  appsettings.Development.json

```
 "ConnectionStrings": {
    "MessageQueue": "amqp://guest:guest@rabbitmq:5672"
  },
```

File: src/docker-compose.yml [hunk 16]
```diff
@@ -13,7 +13,15 @@ services:
     depends_on:
       - rabbitmq
       - db
-
+  delivery:
+    image: ${DOCKER_REGISTRY-}dlselivery
```
Comment (MohammadMokhtarii) [inline src/docker-compose.yml:17]: dlselivery means?!

**Review by MohammadMokhtarii — APPROVED** [review]

**Pull request merged**